src/Pact/Types/Term.hs

```diff
@@ -270,10 +282,19 @@ instance Semigroup Gas where
 instance Monoid Gas where
   mempty = 0
 
-data NativeDFun = NativeDFun {
-      _nativeName :: NativeDefName,
-      _nativeFun :: forall m . Monad m => FunApp -> [Term Ref] -> m (Gas,Term Name)
-    }
+data NativeDFun = NativeDFun
+  { _nativeName :: NativeDefName
+  , _nativeFun :: forall m . Monad m => FunApp -> [Term Ref] -> m (Gas,Term Name)
+  } deriving (Typeable)
+
+nativeDFunDataType :: DataType
+nativeDFunDataType = mkDataType "Pact.Types.Term.NativeDFun" [mkConstr nativeDFunDataType "NativeDFun" [] Prefix]
```

Comment:
@slpopejoy Because we're tying the knot and using it more than once, it's probably best to leave this as a standalone function, rather than declare it twice in the same `Data` instance for `NativeDFun`.  It's long and takes up a lot of unnecessary space that way. I've pared down and gotten rid of the `Constr` declaration, since we only need the datatype info.